compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -1268,23 +1282,30 @@ where
                 }
             };
 
-        if let Certainty::Maybe {
-            cause: cause @ MaybeCause::Overflow { keep_constraints: false, .. },
-            opaque_types_jank,
-        } = certainty
-        {
-            // If we have overflow, it's probable that we're substituting a type
-            // into itself infinitely and any partial substitutions in the query
-            // response are probably not useful anyways, so just return an empty
-            // query response.
-            //
-            // This may prevent us from potentially useful inference, e.g.
-            // 2 candidates, one ambiguous and one overflow, which both
-            // have the same inference constraints.
-            //
-            // Changing this to retain some constraints in the future
-            // won't be a breaking change, so this is good enough for now.
-            return Ok(self.make_ambiguous_response_no_constraints(cause, opaque_types_jank));
+        match self.current_goal_kind {
+            CurrentGoalKind::AliasRelate | CurrentGoalKind::NormalizesTo => {}
```

Comment:
right so here you'd have previously discarded (since `AliasRelate` got `CurrentGoalKind::Misc` but now you take this top branch). I'm not 100% sure why you also go here for `NormalizesTo`. That'd have previously discarded.